From f07e2ffe6d2184ed829d2a2aeda864f02e394986 Mon Sep 17 00:00:00 2001
From: Yang Yingliang <yangyingliang@huawei.com>
Date: Fri, 26 Nov 2021 16:26:50 +0800
Subject: [PATCH] iscsi: add member for NUMA aware order workqueue
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f07e2ffe6d2184ed829d2a2aeda864f02e394986


euleros inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4IZNO
CVE: NA

-------------------------------------------------

Add member to struct iscsi_conn.

Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-By: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: BiaoXiang Ye <yebiaoxiang@huawei.com>
Reviewed-by: fang yi <eric.fangyi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/workqueue.h | 1 +
 include/scsi/libiscsi.h   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/linux/workqueue.h b/include/linux/workqueue.h
index 26de0cae2a0a..0c35ad697a7b 100644
--- a/include/linux/workqueue.h
+++ b/include/linux/workqueue.h
@@ -344,6 +344,7 @@ enum {
 	__WQ_ORDERED		= 1 << 17, /* internal: workqueue is ordered */
 	__WQ_LEGACY		= 1 << 18, /* internal: create*_workqueue() */
 	__WQ_ORDERED_EXPLICIT	= 1 << 19, /* internal: alloc_ordered_workqueue() */
+	__WQ_DYNAMIC		= 1 << 25, /* internal: only support single work order WQ */
 
 	WQ_MAX_ACTIVE		= 512,	  /* I like 512, better ideas? */
 	WQ_MAX_UNBOUND_PER_CPU	= 4,	  /* 4 * #cpus for unbound wq */
diff --git a/include/scsi/libiscsi.h b/include/scsi/libiscsi.h
index e265b274f793..b47428d86a4b 100644
--- a/include/scsi/libiscsi.h
+++ b/include/scsi/libiscsi.h
@@ -237,6 +237,7 @@ struct iscsi_conn {
 	/* custom statistics */
 	uint32_t		eh_abort_cnt;
 	uint32_t		fmr_unalign_cnt;
+	int			intimate_cpu;
 };
 
 struct iscsi_pool {
-- 
2.26.2

